leftwm/src/bin/leftwm-worker.rs

```diff
@@ -21,13 +38,35 @@ fn main() {
         #[cfg(not(feature = "lefthk"))]
         let config = leftwm::load();
 
-        let manager = Manager::<leftwm::Config, XlibDisplayServer>::new(config);
-        manager.register_child_hook();
-        rt.block_on(manager.start_event_loop())
+        match args.get(1) {
+            #[cfg(feature = "xlib")]
+            Some(name) if name == "xlib" => {
+                let manager = Manager::<leftwm::Config, XlibDisplayServer>::new(config);
+
+                manager.register_child_hook();
+                //TODO: Error handling
+                rt.block_on(manager.start_event_loop())
```

Comment:
I think you can take this out of the match condition by defining `let manager;` before or returning it from the statement.
That isn't really necessary now, but if you want error handling it would certainly be duplicated code.

Reply:
It's not possible because `manager` has a different type on each branch. One is `Manager<leftwm::Config, XlibDisplayServer>`, the other one is `Manager<leftwm::Config, X11rbDisplayServer>`